### PR TITLE
Use HOMESTEAD_HOME to define configuration location

### PR DESCRIPTION
**Pull request by nickknissen**
Use HOMESTEAD_HOME enviroment variable instead of HOME so it's possible to move homestead configuration to another location.

### CODE REVIEW

File: homestead [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env php
 <?php
 
-$_ENV['HOME'] = getenv('HOME');
+$_ENV['HOMESTEAD_HOME'] = (getenv('HOMESTEAD_HOME') ? getenv('HOMESTEAD_HOME') : getenv('HOME'));
```
Comment (taylorotwell) [inline homestead]: No need to wrap this in parens.

Comment (nickknissen) [inline homestead]: I removed then parens

**Comment by taylorotwell** [comment]
I'm curious why the variable is put in $_ENV but then referenced out of $_SERVER?

**Comment by nickknissen** [comment]
I more or less just search and replaced. But it properly wold be best to use $_ENV since it's out of request context. I will fix this

File: homestead [hunk 2]
```diff
@@ -15,13 +15,13 @@ else
...
 {
-	if (isset($_SERVER['HOME']))
+	if (isset($_ENV['HOMESTEAD_HOME']))
 	{
-		return $_SERVER['HOME'].'/.homestead';
+		return $_ENV['HOMESTEAD_HOME'].'/.homestead';
 	}
 	else
 	{
-		return $_SERVER['HOMEDRIVE'].$_SERVER['HOMEPATH'].DIRECTORY_SEPARATOR.'.homestead';
```
Comment (taylorotwell) [inline homestead]: These should probably stay $_SERVER since that is what they were before.

Comment (nickknissen) [inline homestead]: But wouldn't `$_ENV` be better since it's a non-server script?

**Comment by 82rules** [comment]
:+1:

**Comment by ThomasBS** [comment]
:+1:

File: homestead (hunk 2 above)
Comment (taylorotwell) [inline homestead]: If it ain't broke don't fix it.

Comment (nickknissen) [inline homestead]: Sure thing :+1:

**Comment by taylorotwell** [comment]
This breaks running "vagrant box update" from the cloned Homestead repository if you are not using the PHP "homestead" helper.

**Pull request merged**